nixos/modules/services/databases/arangodb.nix

```diff
@@ -0,0 +1,63 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.arangodb;
+
+in {
+
+  ###### interface
+
+  options = {
+    services.arangodb = {
+      enable = mkEnableOption "ArangoDB server";
+
+      dataDir = mkOption {
```

Comment:
There are some nice features of `systemd` you can take advantage of if you hardcode this directory to something like `/var/lib/arangodb`, such as `StateDirectory`.

By no means is this a requirement, but recently I'm asking people to question if they actually provide value to users in adding this option. Sometimes the answer is yes, sometimes no. Up to you to decide.